File: rag-quickstart/csharp/src/RagQuickstart/Endpoint.cs
namespace RagQuickstart;

/// <summary>
/// Compose the scheme-prefixed endpoint URL the .NET SDK expects.
/// </summary>
public static class Endpoint
{
    public static string Compose(string host, int port, bool secure)
    {
        var scheme = secure ? "https" : "http";
        return $"{scheme}://{host}:{port}";
    }
}

File: rag-quickstart/typescript/src/endpoint.ts
/**
 * Compose the scheme-prefixed endpoint URL the TS SDK expects.
 *
 * Kept in its own dependency-free module so it is unit-testable without loading
 * the gRPC SDK stack that ermya-client-factory pulls in.
 */

export function composeEndpoint(host: string, port: number, secure: boolean): string {
  const scheme = secure ? "https" : "http";
  return `${scheme}://${host}:${port}`;
}
